plugin.video.composite_for_plex/resources/lib/composite_addon/addon/utils.py

```diff
@@ -0,0 +1,221 @@
+
+    return possible_servers[0]
+
+
+def get_transcode_profile(context):
+    profile_count = 3
+    profile_labels = []
+    append_label = profile_labels.append
+
+    for idx in xrange(profile_count):
+        profile = context.settings.transcode_profile(idx)
+        if profile.get('enabled'):
+            resolution, bitrate = profile.get('quality').split(',')
+            sub_size = profile.get('subtitle_size')
+            audio_boost = profile.get('audio_boost')
+            append_label('[%s] %s@%s (%s/%s)' %
+                         (str(idx + 1), resolution, bitrate.strip(),
+                          sub_size, audio_boost))
+
+    if len(profile_labels) == 1:
+        return 0
+
+    dialog = xbmcgui.Dialog()
+    result = dialog.select(i18n('Transcode Profiles'), profile_labels)
+
+    if result == -1:
+        return 0
+
+    return result
+
+
+def write_pickled(filename, data):
+    try:
+        os.makedirs(CONFIG['temp_path'])
+    except:  # pylint: disable=bare-except
+        pass
+    filename = os.path.join(CONFIG['temp_path'], filename)
+    pickled_data = pickle.dumps(data, protocol=2)
+    with open(filename, 'wb') as open_file:
+        open_file.write(pickled_data)
+
+
+def read_pickled(filename, delete_after=True):
+    filename = os.path.join(CONFIG['temp_path'], filename)
+    if not os.path.exists(filename):
+        return None
+    with open(filename, 'rb') as open_file:
+        pickled_data = open_file.read()
+    if delete_after:
+        try:
+            os.remove(filename)
+        except:  # pylint: disable=bare-except
+            pass
+    return pickle.loads(pickled_data)
+
+
+def notify_all(encoding, method, data):
+    next_data = json.dumps(data)
+    if not isinstance(next_data, bytes):
+        next_data = next_data.encode('utf-8')
+
+    if encoding == 'base64':
+        from base64 import b64encode  # pylint: disable=import-outside-toplevel
+        data = b64encode(next_data)
+        if PY3:
+            data = data.decode('ascii')
+    elif encoding == 'hex':
+        from binascii import hexlify  # pylint: disable=import-outside-toplevel
+        if PY3:
+            if not isinstance(next_data, bytes):
+                next_data = next_data.encode('utf-8')
+            data = hexlify(next_data).decode('utf-8')
+        else:
+            data = hexlify(next_data)
+
+    jsonrpc_request = {
+        "jsonrpc": "2.0",
+        "id": 1,
+        "method": "JSONRPC.NotifyAll",
+        "params": {
+            "sender": "%s.SIGNAL" % CONFIG['id'],
+            "message": method,
+            "data": [data],
+        }
+    }
+
+    _ = xbmc.executeJSONRPC(json.dumps(jsonrpc_request))
+
+
+def jsonrpc_play(url):
+    jsonrpc_request = {
+        "jsonrpc": "2.0",
+        "id": 1,
+        "method": "player.open",
+        "params": {
+            "item": {
+                "file": url
+            }
+        }
+    }
+
+    _ = xbmc.executeJSONRPC(json.dumps(jsonrpc_request))
```

Comment:
no need to assign to an unused var. Just execute the json rpc method.

Reply:
I removed the local variable